public/app/features/teams/TeamGroupSync.tsx

```diff
@@ -1,93 +1,70 @@
 
+import { TeamGroupDto } from '@grafana/api-clients/rtkq/legacy';
 import { Trans, t } from '@grafana/i18n';
 import { Input, Tooltip, Icon, Button, useTheme2, InlineField, InlineFieldRow } from '@grafana/ui';
 import { SlideDown } from 'app/core/components/Animations/SlideDown';
 import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
 import EmptyListCTA from 'app/core/components/EmptyListCTA/EmptyListCTA';
 import { UpgradeBox, UpgradeContent, UpgradeContentProps } from 'app/core/components/Upgrade/UpgradeBox';
 import { highlightTrial } from 'app/features/admin/utils';
-import { StoreState } from 'app/types/store';
-import { TeamGroup } from 'app/types/teams';
 
-import { addTeamGroup, loadTeamGroups, removeTeamGroup } from './state/actions';
-import { getTeamGroups } from './state/selectors';
+import { useAddTeamGroup, useGetTeamGroups, useRemoveTeamGroup } from './hooks';
 
-function mapStateToProps(state: StoreState) {
-  return {
-    groups: getTeamGroups(state.team),
-  };
-}
-
-const mapDispatchToProps = {
-  loadTeamGroups,
-  addTeamGroup,
-  removeTeamGroup,
-};
-
-interface OwnProps {
+interface Props {
   isReadOnly: boolean;
   teamUid: string;
 }
 
-interface State {
-  isAdding: boolean;
-  newGroupId: string;
-}
-
-const connector = connect(mapStateToProps, mapDispatchToProps);
-export type Props = OwnProps & ConnectedProps<typeof connector>;
-
 const headerTooltip = `Sync LDAP, OAuth or SAML groups with your Grafana teams.`;
 
-export class TeamGroupSync extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { isAdding: false, newGroupId: '' };
-  }
-
-  componentDidMount() {
-    this.fetchTeamGroups();
-  }
+export const TeamGroupSync = ({ isReadOnly, teamUid }: Props) => {
+  const [isAdding, setIsAdding] = useState(false);
+  const [newGroupId, setNewGroupId] = useState('');
+  const styles = getStyles();
 
-  async fetchTeamGroups() {
-    this.props.loadTeamGroups(this.props.teamUid);
-  }
+  const { value: groups = [], retry: fetchTeamGroups } = useGetTeamGroups(teamUid);
+  const [addTeamGroup] = useAddTeamGroup();
+  const [removeTeamGroup] = useRemoveTeamGroup();
 
-  onToggleAdding = () => {
-    this.setState({ isAdding: !this.state.isAdding });
+  const onToggleAdding = () => {
+    setIsAdding(!isAdding);
   };
 
-  onNewGroupIdChanged: FormEventHandler<HTMLInputElement> = (event) => {
-    this.setState({ newGroupId: event.currentTarget.value });
+  const onNewGroupIdChanged: FormEventHandler<HTMLInputElement> = (event) => {
+    setNewGroupId(event.currentTarget.value);
   };
 
-  onAddGroup: FormEventHandler<HTMLFormElement> = (event) => {
+  const onAddGroup: FormEventHandler<HTMLFormElement> = async (event) => {
     event.preventDefault();
-    this.props.addTeamGroup(this.props.teamUid, this.state.newGroupId);
-    this.setState({ isAdding: false, newGroupId: '' });
+    await addTeamGroup(teamUid, newGroupId);
+    fetchTeamGroups();
```

Comment:
You shouldn't have to refetch the groups after adding or removing, as the RTK Query mutation should invalidate the tags, so it tells the UI to refetch from the necessary endpoints